RealmSwift/Tests/TestUtils.swift

```diff
@@ -35,6 +35,10 @@ public class Locked<T>: @unchecked Sendable {
         lock.initialize(to: os_unfair_lock())
     }
 
```

Comment:
I would love for this property wrappers to have some sort of documentation/comment, so we can have it mind to use it for testing in new tests.

Reply:
Also is the projected value workaround needed anymore here?

Reply:
There's still a few spots using it where the callback actually will be called on the main thread.